atomic_reactor/plugins/pre_add_filesystem.py

```diff
@@ -374,6 +330,29 @@ def update_repos_from_composes(self):
             self.log.info('adding repo file from compose: %s', compose_info['result_repofile'])
             self.repos.append(compose_info['result_repofile'])
 
+    def _add_filesystem_to_dockerfile(self, file_name, file_path, build_dir: BuildDir):
+        """
+        Put an ADD instruction into the Dockerfile (to include the filesystem
+        into the container image to be built)
+        """
+        content = 'ADD {0} {1}'.format(file_path, file_name)
+        lines = build_dir.dockerfile.lines
+
+        lines[0] = 'FROM scratch\n'
```

Comment:
this is wrong, you are assuming that 1st stage is custom image

but you have to take in account multistage builds, WHERE multiple stages in any place can be custom image

so you have to basically find all custom image stages and substitute those

Reply:
Oh, there can be multiple custom base images? :open_mouth:

Reply:
also keep in mind that you are suppose to change just the image for scratch, so this should still work:

FROM koji/image-build AS importedimage

Reply:
made the requested changes.